drivers/net/nsos_sockets.c

```diff
@@ -258,13 +258,21 @@ static ssize_t nsos_write(void *obj, const void *buf, size_t sz)
 static int nsos_close(void *obj)
 {
 	struct nsos_socket *sock = obj;
+	struct nsos_socket_poll *poll;
 	int ret;
 
 	ret = nsi_host_close(sock->poll.mid.fd);
 	if (ret < 0) {
 		errno = nsos_adapt_get_zephyr_errno();
 	}
 
+	SYS_DLIST_FOR_EACH_CONTAINER(&nsos_polls, poll, node) {
+		if (poll == &sock->poll) {
+			poll->mid.revents = ZSOCK_POLLHUP;
```

Comment:
Should it become `ZSOCK_POLLERR` if `ret < 0`?

Reply:
Possibly? POSIX socket error conventions are not my strong point and to me all three of the options validly describe the situation, so I will follow any suggestion from someone who sounds confident :)
https://github.com/zephyrproject-rtos/zephyr/blob/b775c0d30ea565ee8ad17c31c55baf2d06e75946/include/zephyr/net/socket.h#L54-L59

Reply:
Good question, just wondering what is the case when the close would fail. The end result is still the same i.e., the socket is not used any more and is effectively closed. So perhaps we can send `POLLHUP` in that case too.

Reply:
They're bit flags, we can add both 🤷